app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -260,10 +268,18 @@ public IdentifierHolder createParticipant(Study study, Set<Roles> callerRoles, S
 
         updateAccountOptionsAndRoles(study, callerRoles, options, account, participant);
         
-        String accountId = accountDao.createAccount(study, account, sendVerifyEmail && study
-                .isEmailVerificationEnabled());
+        boolean sendVerifyEmail = requestSendVerifyEmail && study.isEmailVerificationEnabled();
+        
+        account.setStatus(sendVerifyEmail ? AccountStatus.UNVERIFIED : AccountStatus.ENABLED);
+        
+        String accountId = accountDao.createAccount(study, account);
+
         externalIdService.assignExternalId(study, participant.getExternalId(), account.getHealthCode());
         optionsService.setAllOptions(study.getStudyIdentifier(), account.getHealthCode(), options);
+        // send verify email
+        if (sendVerifyEmail) {
+            accountWorkflowService.sendEmailVerificationToken(study, accountId, account.getEmail());
+        }
```

Comment:
Is it worth adding unit tests for (1) requestSendVerifyEmail=false (2)requestSendVerifyEmail=true and (3) requestSendVerifyEmail=true but isEmailVerificationEnabled=false?

Reply:
Added 4 tests specifically for this. Only the case where email verification is enabled, and the caller wants email verification, calls the workflow service (and account status is tied 1:1 to this, account is only unverified if we're going to send the email).